app/models/host/managed.rb

```diff
@@ -955,6 +955,22 @@ def recreate_config
     result
   end
 
+  # converts a name into ip address using DNS.
+  # if we are managing DNS, we can query the correct DNS server
+  # otherwise, use normal systems dns settings to resolv
+  def to_ip_address(name_or_ip)
```

Comment:
Is moving this related to the delegation of public methods?  http://projects.theforeman.org/issues/12067